CI failure in integration tests due to traceback:

```
Traceback (most recent call last):
  File "/tmp/ansible_i9pK4R/ansible_module_azure_rm_keyvault.py", line 435, in <module>
    main()
  File "/tmp/ansible_i9pK4R/ansible_module_azure_rm_keyvault.py", line 432, in main
    AzureRMVaults()
  File "/tmp/ansible_i9pK4R/ansible_module_azure_rm_keyvault.py", line 265, in __init__
    supports_tags=False)
  File "/tmp/ansible_i9pK4R/ansible_modlib.zip/ansible/module_utils/azure_rm_common.py", line 321, in __init__
  File "/tmp/ansible_i9pK4R/ansible_module_azure_rm_keyvault.py", line 307, in exec_module
    self.adjust_parameters()
AttributeError: 'AzureRMVaults' object has no attribute 'adjust_parameters'
```

--- a/lib/ansible/modules/cloud/azure/azure_rm_keyvault.py
+++ b/lib/ansible/modules/cloud/azure/azure_rm_keyvault.py
@@ -304,8 +304,6 @@ class AzureRMVaults(AzureRMModuleBase):
                 elif key == "create_mode":
                     self.parameters.setdefault("properties", {})["create_mode"] = kwargs[key]
 
-        self.adjust_parameters()
-
         old_response = None
         response = None
 

--- a/lib/ansible/modules/cloud/azure/azure_rm_keyvault.py
+++ b/lib/ansible/modules/cloud/azure/azure_rm_keyvault.py
@@ -242,7 +242,8 @@ class AzureRMVaults(AzureRMModuleBase):
             create_mode=dict(
                 type='str',
                 choices=['recover',
-                         'default']
+                         'default'],
+                default='default'
             ),
             state=dict(
                 type='str',
@@ -284,13 +285,13 @@ class AzureRMVaults(AzureRMModuleBase):
                             policy.setdefault("permissions", {})["keys"] = policy["keys"]
                             policy.pop("keys", None)
                         if 'secrets' in policy:
-                            policy.setdefault("permissions", {})["keys"] = policy["keys"]
+                            policy.setdefault("permissions", {})["secrets"] = policy["secrets"]
                             policy.pop("secrets", None)
                         if 'certificates' in policy:
-                            policy.setdefault("permissions", {})["keys"] = policy["keys"]
+                            policy.setdefault("permissions", {})["certificates"] = policy["certificates"]
                             policy.pop("certificates", None)
                         if 'storage' in policy:
-                            policy.setdefault("permissions", {})["keys"] = policy["keys"]
+                            policy.setdefault("permissions", {})["storage"] = policy["storage"]
                             policy.pop("storage", None)
                     self.parameters.setdefault("properties", {})["access_policies"] = access_policies
                 elif key == "enabled_for_deployment":
@@ -329,7 +330,53 @@ class AzureRMVaults(AzureRMModuleBase):
                 self.to_do = Actions.Delete
             elif self.state == 'present':
                 self.log("Need to check if Key Vault instance has to be deleted or may be updated")
-                self.to_do = Actions.Update
+                if ('location' in self.parameters) and (self.parameters['location'] != old_response['location']):
+                    self.to_do = Actions.Update
+                elif ('tenant_id' in self.parameters) and (self.parameters['tenant_id'] != old_response['tenant_id']):
+                    self.to_do = Actions.Update
+                elif ('enabled_for_deployment' in self.parameters) and (self.parameters['enabled_for_deployment'] != old_response['enabled_for_deployment']):
+                    self.to_do = Actions.Update
+                elif (('enabled_for_disk_encryption' in self.parameters) and
+                        (self.parameters['enabled_for_deployment'] != old_response['enabled_for_deployment'])):
+                    self.to_do = Actions.Update
+                elif (('enabled_for_template_deployment' in self.parameters) and
+                        (self.parameters['enabled_for_template_deployment'] != old_response['enabled_for_template_deployment'])):
+                    self.to_do = Actions.Update
+                elif ('enable_soft_delete' in self.parameters) and (self.parameters['enabled_soft_delete'] != old_response['enable_soft_delete']):
+                    self.to_do = Actions.Update
+                elif ('create_mode' in self.parameters) and (self.parameters['create_mode'] != old_response['create_mode']):
+                    self.to_do = Actions.Update
+                # SKU Family
+                # SKU Name
+                # access_policies
+                elif 'access_policies' in self.parameters:
+                    if len(self.parameters['access_policies']) != len(old_response['access_policies']):
+                        self.to_do = Actions.Update
+                    else:
+                        for i in range(len(old_response['access_policies'])):
+                            n = self.parameters['access_policies'][i]
+                            o = old_response['access_policies'][i]
+                            if n.get('tenant_id', False) != o.get('tenant_id', False):
+                                self.to_do = Actions.Update
+                                break
+                            if n.get('object_id', False) != o.get('object_id', False):
+                                self.to_do = Actions.Update
+                                break
+                            if n.get('application_id', False) != o.get('application_id', False):
+                                self.to_do = Actions.Update
+                                break
+                            if n.get('keys', []).cmp(o.get('keys', [])) != 0:
+                                self.to_do = Actions.Update
+                                break
+                            if n.get('secrets', []).cmp(o.get('secrets', [])) != 0:
+                                self.to_do = Actions.Update
+                                break
+                            if n.get('certificates', []).cmp(o.get('certificates', [])) != 0:
+                                self.to_do = Actions.Update
+                                break
+                            if n.get('storage', []).cmp(o.get('storage', [])) != 0:
+                                self.to_do = Actions.Update
+                                break
 
         if (self.to_do == Actions.Create) or (self.to_do == Actions.Update):
             self.log("Need to Create / Update the Key Vault instance")

--- a/lib/ansible/modules/cloud/azure/azure_rm_keyvault.py
+++ b/lib/ansible/modules/cloud/azure/azure_rm_keyvault.py
@@ -33,7 +33,7 @@ options:
     location:
         description:
             - Resource location. If not set, location from the resource group will be used as default.
-    tenant_id:
+    vault_tenant:
         description:
             - The Azure Active Directory tenant ID that should be used for authenticating requests to the key vault.
     sku:
@@ -53,8 +53,8 @@ options:
                     - 'premium'
     access_policies:
         description:
-            - "An array of 0 to 16 identities that have access to the key vault. All identities in the array must use the same tenant ID as the key vault's t
-              enant ID."
+            - "An array of 0 to 16 identities that have access to the key vault. All identities in the array must use the same tenant ID as the key vault's
+               tenant ID."
         suboptions:
             tenant_id:
                 description:
@@ -165,7 +165,7 @@ EXAMPLES = '''
       resource_group: myresourcegroup
       vault_name: samplekeyvault
       enabled_for_deployment: yes
-      tenant_id: 72f98888-8666-4144-9199-2d7cd0111111
+      vault_tenant: 72f98888-8666-4144-9199-2d7cd0111111
       sku:
         name: standard
       access_policies:
@@ -218,7 +218,7 @@ class AzureRMVaults(AzureRMModuleBase):
             location=dict(
                 type='str'
             ),
-            tenant_id=dict(
+            vault_tenant=dict(
                 type='str'
             ),
             sku=dict(
@@ -273,7 +273,7 @@ class AzureRMVaults(AzureRMModuleBase):
             elif kwargs[key] is not None:
                 if key == "location":
                     self.parameters["location"] = kwargs[key]
-                elif key == "tenant_id":
+                elif key == "vault_tenant":
                     self.parameters.setdefault("properties", {})["tenant_id"] = kwargs[key]
                 elif key == "sku":
                     self.parameters.setdefault("properties", {})["sku"] = kwargs[key]

--- a/test/integration/targets/azure_rm_keyvault/tasks/main.yml
+++ b/test/integration/targets/azure_rm_keyvault/tasks/main.yml
@@ -8,7 +8,7 @@
     resource_group: "{{ resource_group }}"
     vault_name: "vault{{ rpfx }}"
     enabled_for_deployment: yes
-    tenant_id: 72f98888-8666-4144-9199-2d7cd0111111
+    vault_tenant: 72f98888-8666-4144-9199-2d7cd0111111
     sku:
       name: standard
       family: xxxx
@@ -26,7 +26,7 @@
     resource_group: "{{ resource_group }}"
     vault_name: "vault{{ rpfx }}"
     enabled_for_deployment: yes
-    tenant_id: 72f98888-8666-4144-9199-2d7cd0111111
+    vault_tenant: 72f98888-8666-4144-9199-2d7cd0111111
     sku:
       name: standard
       family: xxxx
@@ -47,7 +47,7 @@
     resource_group: "{{ resource_group }}"
     vault_name: "vault{{ rpfx }}"
     enabled_for_deployment: yes
-    tenant_id: 72f98888-8666-4144-9199-2d7cd0111111
+    vault_tenant: 72f98888-8666-4144-9199-2d7cd0111111
     sku:
       name: standard
       family: xxxx

